js/views/higher-order/withAutoHide.js

```diff
@@ -0,0 +1,201 @@
+      // for mobile or desktop fullscreen, hide control bar after 3 seconds
+      if (this.props.controller.state.isMobile || this.props.fullscreen || this.props.controller.state.browserSupportsTouch) {
+        this.startHideControlBarTimer();
+      }
+    }
+
+    componentWillUnmount() {
+      document.removeEventListener('mousemove', this.handlePlayerMouseMove);
+      document.removeEventListener('touchmove', this.handlePlayerMouseMove);
+      this.cancelHideControlBarTimer();
+    }
+
+    componentWillReceiveProps(nextProps) {
+      if (nextProps.componentWidth !== this.props.componentWidth) {
+        this.handleResize(nextProps);
+      }
+    }
+
+    componentWillUpdate(nextProps) {
+      if (nextProps) {
+        if (!this.props.fullscreen && nextProps.fullscreen) {
+          this.startHideControlBarTimer();
+        }
+        if (this.props.fullscreen && !nextProps.fullscreen && this.props.controller.state.isMobile) {
+          this.props.controller.showControlBar();
+          this.startHideControlBarTimer();
+        }
+      }
+    }
+
+    /**
+     * Handles the touchEnd event on the Auto Hide Screen.
+     * @private
+     * @param {Event} event The touchEnd event object
+     */
+    handleTouchEnd(event) {
+      if (!this.props.controller.state.controlBarVisible) {
+        this.showControlBar(event);
+        //TODO: Address an existing issue where we don't cancel the timer upon touching control buttons
+        this.startHideControlBarTimer();
+      }
+    }
+
+    /**
+     * Handles the mouseMove and touchMove events.
+     * @private
+     */
+    handlePlayerMouseMove() {
+      if (!this.props.controller.state.isMobile && this.props.fullscreen) {
+        this.showControlBar();
+        this.startHideControlBarTimer();
+      }
+    }
+
+    /**
+     * Handles when the player resizes.
+     * @private
+     */
+    handleResize() {
+      this.startHideControlBarTimer();
+    }
+
+    /**
+     * Handles when a keyboard key is pressed.
+     * @private
+     * @param event The keyDown event object
+     */
+    handleKeyDown(event) {
+      // Show control bar when any of the following keys are pressed:
+      // - Tab: Focus on next control
+      // - Space/Enter: Press active control
+      // - Arrow keys: Either seek forward/back, volume up/down or interact with focused slider
+      switch (event.key) {
+        case CONSTANTS.KEY_VALUES.TAB:
+        case CONSTANTS.KEY_VALUES.SPACE:
+        case CONSTANTS.KEY_VALUES.ENTER:
+        case CONSTANTS.KEY_VALUES.ARROW_UP:
+        case CONSTANTS.KEY_VALUES.ARROW_RIGHT:
+        case CONSTANTS.KEY_VALUES.ARROW_DOWN:
+        case CONSTANTS.KEY_VALUES.ARROW_LEFT:
+          this.showControlBar();
+          this.startHideControlBarTimer();
+          break;
+        default:
+          break;
+      }
+    }
+
+    /**
+     * Handles the mouseout event.
+     * @private
+     * @param event The mouseout event object
```

Comment:
Your param labeling is inconsistent. I think we need the {Event} on all these.  You just have the var name and desc.